### ISSUE TITLE
authentication required to publish ?

### ISSUE THREAD

**Issue by meganewing** [RobertsLab/resources#1716]
Trying to publish my lab notebook to github and keep getting "requires authentication" to publish message with no further indication of how to authenticate. Any advice?

**Comment by kubu4** [comment]
Can you please provide a bit more info, regarding how you're trying to publish to GitHub (e.g. via RStudio Server? via command line? via GitHub Desktop?). That will help us troubleshoot.

And, if you know how, a quick screen capture of the process would also be helpful.

Finally, can you please provide the the exact error message(s) you're getting? Screenshot and/or copy/paste would be great.

**Comment by meganewing** [comment]
Via GitHub Desktop.

I'm following the tutorial [here](https://sr320.github.io/course-fish521-2023/lectures/week_07/lec_06b_websites.html) 

Here's a screenshot of the message I'm getting
<img width="448" alt="Screen Shot 2023-10-02 at 3 11 22 PM" src="https://github.com/RobertsLab/resources/assets/66333757/66da560b-b505-4e66-bd8c-587652035baf">

Other info
- I have 2FA setup per GitHubs new(ish) requirements via authenticator app (DUO) and security keys
- I do have the recovery codes downloaded as well so I thought all was good in terms of my account being authenticated (or "authenticateable" )
- Had my R class instructor and TA who both use github take a look at it today too just in case it was something blaringly obvious but both of them were fairly stumped also
- Running macOS Big Sur 11.2, version Version 3.3.3 (x64) of GitHub Desktop, and Version 2023.09.0+463 of R Studio

I'm happy to try and publish it via other methods if needed

**Comment by kubu4** [comment]
Unrelated, but you should probably uncheck the `Keep this code private" box (free accounts are limited in the number of private repos they can have).

Back on topic, try following this guide:

https://docs.github.com/en/desktop/installing-and-authenticating-to-github-desktop/authenticating-to-github-in-github-desktop?platform=mac

TL;DR - Make sure you're logged into your GitHub account in the GitHub Desktop app.

If we can't resolve this via this issue, maybe lab meeting would be a good time to walk through the whole process in person. Or, Science Hour is exactly suited towards these types of discussions/problems.

**Comment by meganewing** [comment]
I was signed in, but I just signed out and signed back in and bam. Problem solved. 

Must've been signed in from pre-authenticator set up or something. Either way, thanks y'all!
